sqlhooks.go

```diff
@@ -9,12 +9,27 @@ import (
 // Hook is the hook callback signature
 type Hook func(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 
+// ErrorHandler is the error handling callback signature
+type ErrorHandler func(ctx context.Context, err error, query string, args ...interface{}) error
+
 // Hooks instances may be passed to Wrap() to define an instrumented driver
 type Hooks interface {
 	Before(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 	After(ctx context.Context, query string, args ...interface{}) (context.Context, error)
 }
 
+// ErrHook instances will be called if any error happens
+type ErrHook interface {
```

Comment:
The idiomatic name for an interface with a single function is the function name with `er` suffix. EG `fmt.Stringer` has a `String` function. So in this case the interface name would be `OnErrorer`.